nx/lib/nx.ex

```diff
@@ -9377,6 +9377,112 @@ defmodule Nx do
     if opts[:keep_axis], do: res, else: squeeze(res, axes: [axis])
   end
 
+  @doc """
+  Integrate `y` along the given axis using the composite trapezoidal rule.
+  The integration happens in sequence along elements of `x`.
+
+  ## Parameters
+
+      * `axis` - Axis along which to integrate. Defaults to `-1`.
+
+  ## Examples
+
+      iex> Nx.trapz(Nx.tensor([1, 2, 3]), Nx.tensor([4, 5, 6]))
+      #Nx.Tensor<
+        f32
+        4.0
+      >
+
+      iex> Nx.trapz(Nx.tensor([[0, 1, 2], [3, 4, 5]]), Nx.tensor([[1, 2, 3], [1, 2, 3]]))
+      #Nx.Tensor<
+        f32[2]
+        [2.0, 8.0]
+      >
+
+      iex> Nx.trapz(Nx.tensor([[0, 1, 2], [3, 4, 5]]), Nx.tensor([[1, 1, 1], [2, 2, 2]]), axis: 0)
+      #Nx.Tensor<
+        f32[3]
+        [1.5, 2.5, 3.5]
+      >
+  """
+  @doc type: :aggregation, from_backend: false
+  def trapz(y, x, opts \\ []) do
```

Comment:
I'd rename to a more explicit name

Reply:
I think the function should receive x and y instead of y and x

Reply:
Hmm, but in general y in this function is a key argument and originally x is just an optional argument

Reply:
FWIW, scipy also has it as `(y, x)`: https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.trapezoid.html#scipy.integrate.trapezoid

Reply:
I still find it really confusing haha
But it makes sense if you're talking about having multiple arities.

Maybe a general `integrate(y, x \\ nil, opts \\ [])` function with `:method` as a mandatory option, then I can see `y` coming first